### DISCUSSION TITLE
전역변수의 문제점은 무엇인가요?

### DISCUSSION THREAD

**Discussion by pipisebastian** [publdaze/js-study#58] [전역 변수의 문제점]
- 전역변수의 문제점은 무엇인가요?

**Comment by pipisebastian** [comment]
> 전역변수의 문제점은 무엇인가요?

- **암묵적 결합** 의도치 않은 재할당, 어디서든 변수 참조가능.
- 생명주기가 길기에 메모리 차지.
- 스코프상에서 제일 마지막으로 검색.
- **네임스페이스 오염** 파일이 분리되어있더라도, 하나의 전역 스코프를 공유하기 때문에, 각 파일에서 동일한 이름의 변수를 선언한다면, 값이 덮어씌게 되면서 의도치 않은 결과를 일으킵니다.

**Comment by publdaze** [comment]
> - 전역변수의 문제점은 무엇인가요?
- 암묵적 결합 - 어디서든 참조하고 할당할 수 있는 암묵적 결합을 허용하기 때문에 의도치 않은 재할당이 일어날 수 있다.
- 네임스페이스 오염 - 파일이 분리되어 있어도 같은 전역 스코프를 공유하기 때문에 동일한 이름으로 명명된 식별자가 존재할 경우 예상치 못한 결과가 나올 수 있다.
- 스코프 체인 종점에 존재 - 스코프 체인의 종점에 위치하므로 검색 속도가 느리다.
- 긴 생명 주기 - 생명주기가 길어서 생명 주기 동안 메모리 리소스를 오래 잡아먹는다.
